matplotlib-stubs/pyplot.pyi

```diff
@@ -245,9 +246,63 @@ class Figure:
     def savefig(
         self,
         fname: Union[str, Path],
-        dpi: int = ...,
+        dpi: Optional[int] = ...,
+        quality: Optional[int] = ...,
+        optimize: bool = ...,
+        progressive: bool = ...,
+        facecolor: Optional[str] = ...,
+        edgecolor: Optional[str] = ...,
+        orientation: Literal["landscape", "portrait"] = ...,
+        papertype: Literal[
+            "letter",
+            "legal",
+            "executive",
+            "ledger",
+            "a0",
+            "a1",
+            "a2",
+            "a3",
+            "a4",
+            "a5",
+            "a6",
+            "a7",
+            "a8",
+            "a9",
+            "a10",
+            "b0",
+            "b1",
+            "b2",
+            "b3",
+            "b4",
+            "b5",
+            "b6",
+            "b7",
+            "b8",
+            "b9",
+            "b10",
+        ] = ...,
+        format: Literal[
+            "eps",
+            "jpeg",
+            "jpg",
+            "pdf",
+            "pgf",
+            "png",
+            "ps",
+            "raw",
+            "rgba",
+            "svg",
+            "svgz",
+            "tif",
+            "tiff",
+        ] = ...,
+        transparent: bool = ...,
+        bbox_inches: Optional[Union[Literal["tight"], Bbox]] = ...,
+        pad_inches: Optional[int] = ...,
         bbox_extra_artists: Sequence[Artist] = ...,
-        bbox_inches: Optional[Literal["tight"]] = ...,
+        metadata: Union[Dict[str, Optional[str]], Dict[ByteString, ByteString]] = ...,
```

Comment:
is it really `ByteString`? Can it not be a string? Are you maybe looking for [AnyStr](https://docs.python.org/3/library/typing.html#typing.AnyStr)?

Reply:
Really the metadata keys should be a literal but I cba at the moment define a literal for each of the backends.  The documentation for the metadata arg for the pdf backend reads as follows: "Metadata in the PNG file as key-value pairs of bytes or latin-1 encodable strings". For the other two backends string key-value pairs are expected, so far as I can tell.